core/src/main/java/org/apache/iceberg/ManifestMergeManager.java

```diff
@@ -148,11 +167,13 @@ private Iterable<ManifestFile> mergeGroup(
               // if the number of manifests is above the minimum count. this is applied only to bins
               // with an in-memory
               // manifest so that large manifests don't prevent merging older groups.
-              if (bin.contains(first) && bin.size() < minCountToMerge) {
+              if ((bin.contains(first) && bin.size() < minCountToMerge)
+                  || bin.size() > mergeLimit.get()) {
                 // not enough to merge, add all manifest files to the output list
                 outputManifests.addAll(bin);
               } else {
                 // merge the group
+                mergeLimit.addAndGet(-1 * bin.size());
```

Comment:
just curious, in what order are we consuming this limit? latest commit time desc ?

Reply:
The order is not guaranteed since it is processed concurrently.